board/rriv_board_0_4_2/src/lib.rs

```diff
@@ -53,57 +69,107 @@ pub struct Serial {
+        let device_peripherals = pac::Peripherals::take().unwrap();
+        // rprintln!("{:?}",device_peripherals.RCC.apb2enr); //.i2c1en();  //enabled();
+        // device_peripherals.RCC.apb2enr.
 
         // Take ownership over the raw flash and rcc devices and convert them into the corresponding
         // HAL structs
-        let mut flash = p.FLASH.constrain();
-        let rcc = p.RCC.constrain();
+        let mut flash = device_peripherals.FLASH.constrain();
+        let rcc = device_peripherals.RCC.constrain();
+        let mut afio = device_peripherals.AFIO.constrain();     // Prepare the alternate function I/O registers
+
+
+        // Prepare the GPIO
+        let mut gpioa = device_peripherals.GPIOA.split();
+        let mut gpiob = device_peripherals.GPIOB.split();
+        let mut gpioc = device_peripherals.GPIOC.split();
+        let mut gpiod = device_peripherals.GPIOD.split();
+
+
+        // Set up pins
+        let mut pins = Pins::build(gpioa, gpiob, gpioc, gpiod);
+        // self.pins = Some(pins);
+        // let pins = self.pins;
+
+ 
+        let mut switched_pow = pins.enable_3v.into_push_pull_output(&mut pins.gpioc.crl);
+        let mut hse_pin = pins.enable_hse.into_push_pull_output(&mut pins.gpioc.crh); // high speed external on/off
+        
+        let wake = pins.wake_button();
+
+        // Turn on the HSE
+        hse_pin.set_high();
 
         // Freeze the configuration of all the clocks in the system
         // and store the frozen frequencies in `clocks`
         let clocks = rcc
-        .cfgr
-        .use_hse(8.MHz())
-        .sysclk(48.MHz())
-        .pclk1(24.MHz())
-        .freeze(&mut flash.acr);
+            .cfgr
+            .use_hse(8.MHz())
+            .sysclk(48.MHz())
+            .pclk1(24.MHz()) // was 24, set to 6 following i2c code
+            .freeze(&mut flash.acr);
 
         assert!(clocks.usbclk_valid());
 
-
-        // Prepare the alternate function I/O registers
-        let mut afio = p.AFIO.constrain();
-
-        // Prepare the peripherals
-        let mut gpioa = p.GPIOA.split();
-        let mut gpioc = p.GPIOC.split();
-
-        // USART2
-        let tx_pin = gpioa.pa2.into_alternate_push_pull(&mut gpioa.crl);
-        let rx_pin = gpioa.pa3;
+  
+        // Set up and initialize switched power bus
+        self.delay = Some(core_peripherals.SYST.delay(&clocks));
+        switched_pow.set_high();
+        self.delay_ms(250_u16);
+        switched_pow.set_low();
+        self.delay_ms(250_u16);
+        switched_pow.set_high();
+        self.delay_ms(250_u16);
+        switched_pow.set_low();
+        self.delay_ms(250_u16);
+    
 
         // Init Serial
         // rprintln!("initializing serial");
+        let tx_pin = pins.tx.into_alternate_push_pull(&mut gpioa.crl); // USART2
+        let rx_pin = pins.rx; // USART2
+
         let mut serial = Hal_Serial::new(
-            p.USART2,
+            device_peripherals.USART2,
```

Comment:
is this what rust embedded community likes to name 'pac' for peripheral access crate? idk, asking for a friend. 🕵️